pkg/lib/hook/deno_client.go

```diff
@@ -15,37 +15,53 @@ import (
 
 //go:generate mockgen -source=deno_client.go -destination=deno_client_mock_test.go -package hook
 
-type SyncDenoClient interface {
+type DenoClient interface {
 	Run(ctx context.Context, script string, input interface{}) (out interface{}, err error)
 }
 
+type SyncDenoClient interface {
+	DenoClient
+}
+
 func NewSyncDenoClient(endpoint config.DenoEndpoint, c *config.HookConfig, logger Logger) SyncDenoClient {
-	return &DenoClient{
+	return &DenoClientImpl{
 		Endpoint:   string(endpoint),
 		HTTPClient: httputil.NewExternalClient(c.SyncTimeout.Duration()),
 		Logger:     logger,
 	}
 }
 
 type AsyncDenoClient interface {
-	Run(ctx context.Context, script string, input interface{}) (out interface{}, err error)
+	DenoClient
 }
 
 func NewAsyncDenoClient(endpoint config.DenoEndpoint, logger Logger) AsyncDenoClient {
-	return &DenoClient{
+	return &DenoClientImpl{
 		Endpoint:   string(endpoint),
 		HTTPClient: httputil.NewExternalClient(60 * time.Second),
 		Logger:     logger,
 	}
 }
 
-type DenoClient struct {
+type DenoClientFactory func(timeout time.Duration) DenoClient
```

Comment:
This approach of supporting timeout complicates the code a lot. As mention in my previous comment, we now have 

```golang
func (h *WebHookImpl) PerformNoResponse(client *http.Client, request *http.Request) (error)
```

We can easily pass in a HTTP client configured with suitable timeout. The HTTP client should be dependency injected into CustomClient, like what we did for hook.SyncHTTPClient and hook.AsyncHTTPClient.

---

So the question becomes how can we do the same for Deno.

It turns out that CustomClient only needs `loadScript` of DenoHookImpl. CustomClient can create its own DenoClient to execute the script. So we only need to expose LoadScript and dependency-inject a DenoClient with suitable timeout.

```golang
type DenoScriptLoader interface {
  LoadScript(u *url.URL) ([]byte, error)
}

type DenoClient interface {
  	Run(ctx context.Context, script string, input interface{}) (out interface{}, err error)
}

func NewDenoClient() DenoClient {
  return &hook.DenoClient{} // Use the suitable timeout here.
}

type CustomClient struct {
  DenoScriptLoader DenoScriptLoader
  DenoClient DenoClient
}
```

Reply:
Instead of creating `DenoScriptLoader`, I make the client used on run a parameter.
```golang
func (h *DenoHook) Run(client DenoClient, u *url.URL, input interface{}) (out interface{}, err error)
```
You may see the change in 44cdd6b